vm/src/frame.rs

```diff
@@ -715,18 +715,38 @@ impl Frame {
                 Ok(None)
             }
 
-            bytecode::Instruction::Raise { argc } => {
+            bytecode::Instruction::Raise { argc, in_exc } => {
                 let cause = match argc {
                     2 => self.get_exception(vm, true)?,
                     _ => vm.get_none(),
                 };
                 let exception = match argc {
+                    0 => match in_exc {
+                        true => self.get_exception(vm, false)?,
+                        false => {
+                            return Err(vm.new_exception(
+                                vm.ctx.exceptions.runtime_error.clone(),
+                                "No active exception to reraise".to_string(),
+                            ));
+                        }
+                    },
                     1 | 2 => self.get_exception(vm, false)?,
-                    0 | 3 => panic!("Not implemented!"),
+                    3 => panic!("Not implemented!"),
                     _ => panic!("Invalid parameter for RAISE_VARARGS, must be between 0 to 3"),
                 };
-                info!("Exception raised: {:?} with cause: {:?}", exception, cause);
+                let context = match argc {
+                    0 => vm.get_none(), // We have already got the exception,
+                    _ => match in_exc {
```

Comment:
You could just use an if/else here

Reply:
Changed. Not sure if it looks better.

Reply:
Oh sorry I meant for matching on `in_exc`, the line selection for reviews for the GitHub client I'm using is weird.

Reply:
I think it looks better with the match. Would you be fine with it staying as it is?

Reply:
Sure, np